oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -36,39 +36,44 @@
    * @param refreshToken A refresh token resulting from a OAuth2 consent flow.
    */
   public UserCredentials(String clientId, String clientSecret, String refreshToken) {
-    this(clientId, clientSecret, refreshToken, null, null);
+    this(clientId, clientSecret, refreshToken, null, null, null);
   }
 
   /**
-   * Constructor allow override of transport for getting access tokens.
+   * Constructor to allow both refresh token and initial access token for 3LO scenarios.
    *
    * @param clientId Client ID of the credential from the console.
    * @param clientSecret Client ID of the credential from the console.
    * @param refreshToken A refresh token resulting from a OAuth2 consent flow.
-   * @param transport HTTP object used to get acces tokens.
+   * @param accessToken Initial or temporary access token.
    */
   public UserCredentials(
-      String clientId, String clientSecret, String refreshToken, HttpTransport transport) {
-    this(clientId, clientSecret, refreshToken, transport, null);
+      String clientId, String clientSecret, String refreshToken, AccessToken accessToken) {
+    this(clientId, clientSecret, refreshToken, accessToken, null, null);
   }
 
+
   /**
-   * Constructor allowing override of transport and URL for getting access tokens.
+   * Constructor with all parameters allowing custom transport and server URL.
    *
    * @param clientId Client ID of the credential from the console.
    * @param clientSecret Client ID of the credential from the console.
    * @param refreshToken A refresh token resulting from a OAuth2 consent flow.
-   * @param transport HTTP object used to get acces tokens.
-   * @param tokenServerUrl URL of the endpoint that provides tokens.
+   * @param accessToken Initial or temporary access token.
+   * @param transport HTTP object used to get access tokens.
+   * @param tokenServerUrl URL of the end point that provides tokens.
    */
   public UserCredentials(String clientId, String clientSecret, String refreshToken,
-      HttpTransport transport, GenericUrl tokenServerUrl) {
+      AccessToken accessToken, HttpTransport transport, GenericUrl tokenServerUrl) {
+    super(accessToken);
     this.clientId = Preconditions.checkNotNull(clientId);
     this.clientSecret = Preconditions.checkNotNull(clientSecret);
-    this.refreshToken = Preconditions.checkNotNull(refreshToken);
+    this.refreshToken = refreshToken;
```

Comment:
Yes, you can theoretically use a user access token without a client id or secret. I think to enable this, we want a different class, since if you only have an access token, then it is a general purpose temporary credential. To model this I think we should either have a TemporaryCredentials class, or we should change OAuth2Credenthals to be a non-abstract type where you can pass in an AccesToken directly. I was planning to do this in a subsequent change. But for 3LO scenarios, even without a refresh token, you would always have the Client ID and secret on hand because you need it get the access token.